content/docs/how-we-work/baseline/logging-details.md

```diff
@@ -0,0 +1,145 @@
+   - Implement custom logging for complex or critical database operations using hooks and custom logging functions.
+
+3. **API Calls:**
+   - Log all outgoing and incoming API requests and responses. Use the Guzzle middleware to log API requests made through Guzzle HTTP client.
+   - Implement logging in custom API integrations by capturing request and response data and logging it using `\Drupal::logger`.
+
+4. **Third-Party Integrations:**
+   - Ensure that all third-party services and integrations have appropriate logging in place. This can include logging successful and failed requests, responses, and any relevant data exchanges.
+   - Utilize existing logging features provided by third-party modules and extend them as needed to capture additional details.
+
+5. **Error Levels:**
+   - Use appropriate error levels to categorize logs, but avoid sending debug and info logs to Sentry as they are not suitable for these log types.
+   - Set the logging threshold to `Warning`, `Error`, or `Critical` to filter and prioritize issues effectively.
+   - Explicitly configure your logging settings to exclude debug and info logs by adjusting the `severity level` in your Raven module settings or custom logging configuration.
+
+6. **Monitoring and Review:**
+   - Regularly review and monitor logs using the Drupal logging interface or integrate with external monitoring tools like Sentry to ensure comprehensive coverage.
+   - Set up automated log review and alerting processes to quickly identify and address any logging gaps or issues.
+
+### Regularly Review Logs
+
+1. **Log Review Sessions:**
+   - Schedule regular log review sessions to periodically review logs and identify recurring issues as part of your development and maintenance processes.
+
+2. **Log Retention Policies:**
+
+   - Implement log retention policies to manage log storage efficiently. Ensure that logs are archived and accessible for future reference.
+   - Define log retention periods and configure log rotation to prevent excessive log storage and maintain performance.
+
+### Optimize Alert Rules
+
+1. **Alert Configuration:**
+   - Fine-tune alert rules to minimize noise and focus on critical issues by setting appropriate thresholds and conditions for alerts.
+
+2. **Alert Tuning:**
+   - Continuously monitor and adjust alert rules based on feedback and changing application behavior. Engage with the team to refine alert settings.
+
+### Automate Ticket Creation
+
+1. **Integration:**
+   - Automate the ticket creation process to streamline incident management. Use Sentry’s integration capabilities to automatically create tickets for high-priority issues.
+
+2. **Automation Tools:**
+   - Explore automation tools and scripts that can help manage alerts and tickets more efficiently. This can include auto-assigning tickets, updating statuses, and notifying relevant stakeholders.
```

Comment:
This was also covered before, right?

Reply:
@hussainweb You are right, but it is mentioned here as a best practice.